src/Apps/Order/redirects.tsx

```diff
@@ -205,7 +205,6 @@ export const redirects: RedirectRecord<OrderQuery> = {
       path: "payment/new",
       rules: [
         goToStatusIfBuyNowCreditCardOrder,
-        goToStatusIfOrderIsNotSubmitted,
```

Comment:
Thanks for finding it! 

You think the idea here is that we only allow to update payment for the order if payment is failing? Wonder if we should update `goToStatusIfNotLastTransactionFailed` to look at displayState now? This way we are moving a step closer to using one consolidated `state` for those checks?